docs/quantum_chess/concepts.ipynb

```diff
@@ -0,0 +1,2063 @@
+    "\n",
+    "```\n",
+    "|000 010 000> + |000 100 000>\n",
+    "```\n",
+    "\n",
+    "![Quantum king moves](../images/qc_qking_moves.png)\n",
+    "\n",
+    "\n",
+    "Now, suppose that we make two additional moves.  First, we will split the \"half-king\" on 'b1' to 'c1' and 'c2'.  Next, we will split the 'half king' on 'b2' to 'b3' and 'c3'.\n",
+    "\n",
+    "So, despite that we have made three split moves that involve almost all of the squares on the board, we still only have to worry about 4 of the 512 states on the board.  The state is now (ignoring the constant factor):\n",
+    "\n",
+    "```\n",
+    "|000 000 100> + |000 000 010> + |000 001 000> + |000 000 001>\n",
+    "```\n",
+    "\n",
+    "Of course, if we continue splitting moves, we will eventually get to a large number of potential states.  However, we will never need to consider all 512 states in this example, since some states can never be reached by a sequence of moves.  For instance, the state `|110 000 000>` represents a king on a1 *and* a king on b1.  There's no way where we can start the board with one king on the board and suddenly wind up with two kings (unless we have noise, more on that later).\n",
+    "\n",
+    "**Exercise to the reader**: How many basis states are possible to include in a superposition if we allow the player to make as many normal/split moves as they want? How many superpositions are possible?\n",
+    "\n",
+    "**Summary**\n",
+    "\n",
+    "Since the space of possible quantum chess positions is *much* smaller than the space of possible quantum states, it is very tractable to simulate quantum chess if we have a custom simulator tuned to consider only relevant states.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "PMKj8n_Rk_uB"
+   },
+   "source": [
+    "## Quantum Chess in Cirq\n",
+    "\n",
+    "Next, let's define each of these moves in cirq.  We will define a function for each of three possible moves we are considering.  Each of these will use make use of a python structure called [generators](https://wiki.python.org/moin/Generators) which allows us to 'yield' one or more operations.  This will allow us to use these functions very flexibly while we construct circuits.\n",
+    "\n",
+    "In addition, we will define a function for placing a piece on the board.  Cirq simulators initialize into the zero state (an empty board), so we will use an 'X' gate to turn on a qubit and place a piece on the board."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "WXr_a4xri01v"
+   },
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "\n",
+    "def normal_move(s: cirq.Qid, t: cirq.Qid):\n",
+    "  \"\"\" A normal move in quantum chess.\n",
+    "\n",
+    "  This function takes two qubits and returns a generator\n",
```

Comment:
the syntax highlighting in jupyter notebook looks freaky for these functions because they are indented with two spaces (instead of the normal four)